### PR TITLE
Refactor Symfony user deprecations.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ x ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Fix Symfony UserInterface deprecations.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. See if the build completes.

### CODE REVIEW

File: app/bundles/UserBundle/Event/AuthenticationEvent.php [hunk 1]
```diff
@@ -116,7 +116,7 @@ public function setToken($service, TokenInterface $token): void
     {
         $this->token                 = $token;
         $this->authenticatingService = $service;
-        $this->isAuthenticated       = $token->isAuthenticated();
+        $this->isAuthenticated       = null === $token->getUser();
```
Comment (kuzmany) [inline app/bundles/UserBundle/Event/AuthenticationEvent.php:119]: Shouldn't this condition not equal NULL?

Comment (kuzmany) [inline app/bundles/UserBundle/Event/AuthenticationEvent.php]: Shouldn't this condition not equal NULL?

**Review by kuzmany — CHANGES_REQUESTED** [review]

File: app/bundles/UserBundle/Event/AuthenticationEvent.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/UserBundle/Event/AuthenticationEvent.php]: @kuzmany Good find! Fixed.

**Review by kuzmany — APPROVED** [review]
Good work 👍

File: app/bundles/UserBundle/Security/Provider/UserProvider.php [hunk 2]
```diff
@@ -37,6 +37,11 @@ public function __construct(
      * @return User
      */
     public function loadUserByUsername($username)
```
Comment (escopecz) [inline app/bundles/UserBundle/Security/Provider/UserProvider.php:39]: Shall we deprecate the methods that are now just proxies to the new methods?

Comment (biozshock) [reply app/bundles/UserBundle/Security/Provider/UserProvider.php:39]: They are deprecated in the interface already.

**Review by escopecz — APPROVED** [review]
Thank you!

**Pull request merged**